File: apps/api/app/schemas/trend.py
# Defines trend schemas for FastAPI responses.
from typing import Literal

from pydantic import BaseModel

Platform = Literal["instagram-reels", "tiktok", "youtube-shorts"]
TrendCategory = Literal["mukbang", "ai-video", "dance", "beauty", "fashion", "daily"]
TrendDirection = Literal["rising", "stable", "watch"]


class Trend(BaseModel):
    id: str
    title: str
    description: str
    category: TrendCategory
    platforms: list[Platform]
    score: int
    growth_rate: int
    direction: TrendDirection
    tags: list[str]
    predicted_peak: str
    created_at: str

File: apps/web/src/types/trend.ts
// Defines trend-related TypeScript contracts for the web app.
export type Platform = "instagram-reels" | "tiktok" | "youtube-shorts";

export type TrendCategory = "mukbang" | "ai-video" | "dance" | "beauty" | "fashion" | "daily";

export type TrendDirection = "rising" | "stable" | "watch";

export type Trend = {
  id: string;
  title: string;
  description: string;
  category: TrendCategory;
  platforms: Platform[];
  score: number;
  growthRate: number;
  direction: TrendDirection;
  tags: string[];
  predictedPeak: string;
  createdAt: string;
};
